### ISSUE TITLE
Can't converge when trainning using TensorFlow

### ISSUE THREAD

**Issue by chenynCV** [iro-cp/FCRN-DepthPrediction#13]
I am trying to train this model using your TensorFlow code. But it can't converge.
I am using 'nyu_depth_v2_labeled.mat'. I accept the L2 loss for convenience. 
The raw depths are used.  Invalid pixels (where depth is zero) have been excluded from training. I have tried to fix the ResNet50 or not.

The code is as below:
```
# NetWork
graph = tf.Graph()
with graph.as_default():
    # Create a placeholder for the input image
    tf_train_dataset = tf.placeholder(tf.float32, shape=(None, img_height, img_width, num_channels))
    tf_train_labels = tf.placeholder(tf.float32, shape=(None, depth_height, depth_width))

    # Model.
    # net = models.ResNet50UpProj({'data': tf_train_dataset}, batch_size, trainable=True)
    net_ResNet50 = models.ResNet50({'data': tf_train_dataset}, batch_size, trainable=False)
    layer1_BN = net_ResNet50.get_output()
    net = models.UpProj({'layer1_BN': layer1_BN}, batch_size, trainable=True)

    # Training computation.
    output = tf.squeeze(net.get_output(), squeeze_dims=[3])
    loss = tf.reduce_mean(tf.nn.l2_loss((output-tf_train_labels)*(tf_train_labels!=0)))

    # Optimizer.
    global_step = tf.Variable(0, trainable=False)
    starter_learning_rate = 10**-3
    learning_rate = tf.train.exponential_decay(starter_learning_rate, global_step,
                                               200, 0.8, staircase=True)
    # optimizer = tf.train.GradientDescentOptimizer(learning_rate).minimize(loss, global_step=global_step)
    momentum = 0.9
    optimizer = tf.train.MomentumOptimizer(learning_rate, momentum).minimize(loss, global_step=global_step)

    # Add a scalar summary for the snapshot loss.
    tf.summary.scalar('loss', loss)

    # Build the summary Tensor based on the TF collection of Summaries.
    summary = tf.summary.merge_all()

# Train
with tf.Session(graph=graph) as session:
  tf.global_variables_initializer().run()
  saver = tf.train.Saver()

  # Instantiate a SummaryWriter to output summaries and the Graph.
  summary_writer = tf.summary.FileWriter(log_dir, session.graph)

  # Load the converted parameters
  print('Loading the model')
  net.load('NYU_ResNet-UpProj.txt',session)
  print("Initialized")

  for step in range(num_steps):
    # Pick an offset within the training data, which has been randomized.
    # Note: we could use better randomization across epochs.
    offset = (step * batch_size) % (train_labels.shape[0] - batch_size)
    # Generate a minibatch.
    batch_data = train_dataset[offset:(offset + batch_size), :]
    batch_labels = train_labels[offset:(offset + batch_size), :]

    # Prepare a dictionary telling the session where to feed the minibatch.
    # The key of the dictionary is the placeholder node of the graph to be fed,
    # and the value is the numpy array to feed to it.
    feed_dict = {tf_train_dataset : batch_data, tf_train_labels : batch_labels}
    _, l, predictions = session.run([optimizer, loss, output], feed_dict=feed_dict)

    if (step % 10 == 0):
      print("Minibatch loss at step %d: %f" % (step, l))
      print("Minibatch accuracy: %.1f" % accuracy(predictions, batch_labels))
      # Update the events file.
      summary_str = session.run(summary, feed_dict=feed_dict)
      summary_writer.add_summary(summary_str, step)
      summary_writer.flush()

  # print("Test accuracy: %.1f%%" % accuracy(test_prediction.eval(), test_labels))

  saver.save(session, model_path)

  print('Done!!!')
```
The loss curve is as follws:
![loss](https://cloud.githubusercontent.com/assets/8617034/23705013/881965ec-0442-11e7-95a0-61aab6130acf.png)

**Comment by chrirupp** [comment]
Dear chenynCV,

it appears that you use the labelled dataset only and do no augmentations. We found that sampling the 400GB of the raw dataset plus augmentations was crtitical to the training. Check the paper for details.

Best,
Chris

**Comment by chenynCV** [comment]
Year, I have used the labelled dataset only indeeed. I think this process is like fine-tune. It should converge if I use the model right.

**Comment by chrirupp** [comment]
Ah, I see... You are loading our pre-trained model. Maybe the problem is that you do not divide by the number of valid pixels in your L2 loss?

**Comment by chenynCV** [comment]
Thanks for your reply. I will try it later.

**Comment by leonardoaraujosantos** [comment]
Hi @chenynCV this code that you're using for training does not implement the reverse huber loss correct?

**Comment by Flying-Sailor** [comment]
@chenynCV Hi, i am trying to implement this paper, but the depth maps are blurred, wondering how you were able to do it,
any suggestions appreciated, thanks in advance.
